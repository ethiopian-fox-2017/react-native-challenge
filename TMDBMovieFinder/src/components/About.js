import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  Image,
  Button
} from 'react-native'

import Link from './Link'
import { styles } from '../styles'

export default class About extends React.Component {

  render () {
    const { navigation } = this.props
    return (
      <View style={styles.aboutBox}>
        <Image source={require('../img/tmdb1.png')} style={styles.logoTMDB} />
        <Text style={styles.aboutText}>A simple Movie Finder App powered by The Movie Database API to get the latest most popular movie</Text>
        <View>
          <Link url={'https://github.com/andrasat'}/>
        </View>
        <Button
          title="Go Back"
          color="#01d277"
          onPress={() => navigation.goBack()}
        />
      </View>
    )
  }
}

About.propTypes = {
  navigation: PropTypes.object.isRequired
}