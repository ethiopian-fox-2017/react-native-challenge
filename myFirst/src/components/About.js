import React, { Component } from 'react';
import { Image, Linking } from 'react-native';

import { Container, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body, H1 } from 'native-base';


class About extends Component {
  static navigationOptions = {
    title: 'About',

  };
  render() {
    const { navigation } = this.props;
    return(
      <Container style={{ padding: 20 }}>

        <Content>
        <H1 style={{ textAlign: 'center' }}>About Me</H1>
          <Card style={{ flex: 0 }}>
            <CardItem>
              <Left>
                <Image
                  style={{width: 60, height: 60, borderRadius: 100}}
                  source={{uri: 'https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg' }}
                />
                <Body>
                  <Text>Daily Information</Text>
                  <Text note>By: Anthony Juan</Text>
                </Body>
              </Left>
            </CardItem>
            <CardItem cardBody>
              <Image />
            </CardItem>
            <CardItem content>
              <Text>
                Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
              </Text>
            </CardItem>
            <CardItem style={{justifyContent: 'space-around'}}>
            <Button className="buttonGit" transparent onPress={() => {Linking.openURL('https://github.com/anthonyjuan')}}>
              <Image
              style={{width: 30, height: 30, borderRadius: 100}}
              source={{uri:'http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png'}}
              />
            </Button>

            </CardItem>
          </Card>
        </Content>
        <Button iconLeft light onPress={() => navigation.goBack()}>
           <Icon name='arrow-back' />
           <Text>Back</Text>
         </Button>
      </Container>
    )
  }
}

export default About;
