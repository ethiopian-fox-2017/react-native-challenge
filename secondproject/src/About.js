import React from 'react'

import {
  Text,
  View,
  Button,
} from 'react-native'

class About extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>
          This is about
        </Text>
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    backgroundColor: 'teal'
  }
}

export default About