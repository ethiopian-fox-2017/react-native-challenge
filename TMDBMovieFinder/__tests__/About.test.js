import 'react-native'
import React from 'react'

import About from '../src/components/About'
import renderer from 'react-test-renderer'

it('Render About', () => {
  const wrapper = renderer.create(
    <About />
  )
  expect(wrapper).toMatchSnapshot()
})