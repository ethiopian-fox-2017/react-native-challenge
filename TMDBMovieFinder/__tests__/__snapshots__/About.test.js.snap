// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render About 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "height": "100%",
      "justifyContent": "space-around",
      "padding": 5,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 130,
        "width": 150,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "width": "75%",
      }
    }
  >
    A simple Movie Finder App powered by The Movie Database API to get the latest most popular movie
  </Text>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          My Github Page
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#01d277",
            },
          ]
        }
      >
        Go Back
      </Text>
    </View>
  </View>
</View>
`;
